UserTools/DirectionGridCheck/DirectionGridCheck.cpp

```diff
@@ -0,0 +1,245 @@
+                phi -= 2 * TMath::Pi();
+            }
+
+            tempdir.SetTheta(theta);
+            tempdir.SetPhi(phi);
+            AngularPlot->SetPoint(l, phi, theta);
+            std::cout << "Angular Plot set:" << l << ", " << phi << ", " << theta << endl;
+        }
+    }
+    RecoVertex* truevtx = 0;
+    RecoVertex* recovtx = 0;
+    auto get_muonMC = m_data->Stores.at("RecoEvent")->Get("TrueVertex", truevtx);
+    auto get_muonReco = m_data->Stores.at("RecoEvent")->Get("ExtendedVertex", recovtx);
+    double truePhi = truevtx->GetDirection().GetPhi();
+    double trueTheta = truevtx->GetDirection().GetTheta();
+    double RecoDirX = recovtx->GetDirection().X();
+    double RecoDirY = recovtx->GetDirection().Y();
+    double RecoDirZ = recovtx->GetDirection().Z();
+    double recoPhi = atan(RecoDirY/RecoDirX);
+    double recoTheta = asin(RecoDirX * RecoDirX + RecoDirY * RecoDirY);
+    if (trueTheta < 0) trueTheta += 2*TMath::Pi();
+    if (trueTheta > 2 * TMath::Pi()) trueTheta -= 2 * TMath::Pi();
+    TrueDirPlot->SetPoint(eventCount, truePhi, trueTheta);
+    std::cout << "True Plot set: " << eventCount << ", " << truePhi << ", " << trueTheta << endl;
+    Direction MRDDirection = this->findDirectionMRD();
+    MRDDirPlot->SetPoint(eventCount, MRDDirection.GetPhi(), MRDDirection.GetTheta());
+    std::cout << "MRD Plot set: " << eventCount << ", " << MRDDirection.GetPhi() << ", " << MRDDirection.GetTheta() << endl;
+    RecoDirPlot->SetPoint(eventCount, recoPhi, recoTheta);
+    std::cout << "recoPlot set: " << eventCount << ", " << recoPhi << ", " << recoTheta << endl;
+
+    eventCount++;
+  return true;
+}
+
+
+bool DirectionGridCheck::Finalise(){
+    fOutput_tfile->cd();
+    AngularPlot->Write("Angular Grid");
+    TrueDirPlot->Write("trueDirection");
+    MRDDirPlot->Write("MRDDirection");
+    RecoDirPlot->Write("recoDirection");
+    fOutput_tfile->Write();
+    fOutput_tfile->Close();
+  return true;
+}
+
+Direction DirectionGridCheck::findDirectionMRD() {
+	std::vector<BoostStore>* Tracks;
+    int numtracksinev;
+	m_data->Stores["MRDTracks"]->Get("MRDTracks", Tracks);
+	m_data->Stores["MRDTracks"]->Get("NumMrdTracks", numtracksinev);
+
+	if (numtracksinev > 1) Log("Multiple tracks need work; just using first for now", v_debug, verbosity);
+	double gradx, grady, theta, phi;
+	Direction startVertex, endVertex, result;
+	BoostStore* thisTrack = &(Tracks->at(0));
+
+	thisTrack->Get("VTrackGradient", gradx);
+	thisTrack->Get("HTrackGradient", grady);
+	phi = atan(grady / gradx);
+	theta = asin(pow((gradx * gradx + grady * grady), 0.5));
+	/*TRandom3 smear;
+	Direction vtxDir = fTrueVertex->GetDirection();
+	Direction result;
+	result.SetTheta(smear.Gaus(vtxDir.GetTheta(), 0.4));
+	result.SetPhi(smear.Gaus(vtxDir.GetPhi(), 0.4));*/
+	result.SetTheta(theta);
+	result.SetPhi(phi);
+
+	return result;
+}
+
+RecoVertex* DirectionGridCheck::FindSimpleDirection(RecoVertex* myVertex) {
```

Comment:
This function is never used. 
A quick search of the repository shows that there is a `FindSimpleDirection(RecoVertex* myVertex)` method in the `VtxSeedFineGrid` class, the `VtxPointDirectionFinder` class, the `VtxExtendedVertexFinder` class, *and* now this class. A cursory look suggests this is the same code in all places, although whether they are exactly the same in all cases I haven't checked. This is **very bad practice** - if there's a bug in the `FindSimpleDirection` method, it would need to be fixed in 3/4 different locations! Please centralise this! (and remove it from here if it's not needed).

Reply:
The readme for the `VtxExtendedVertexFinder` Tool in fact states that
> For each position, a direction seed is generated using the "FindSimpleDirection" tool.
but there is no such Tool. Perhaps there should be.